### DISCUSSION TITLE
Can root dir not have a `meson.build`

### DISCUSSION THREAD

**Discussion by Freed-Wu** [JCWasmx86/Swift-MesonLSP#76] [Q&A]
For example, current `$PWD` is `/the/path`, there is not `/the/path/meson.build`. I try to edit `/the/path/subdir/meson.build`, mesonlsp will not work unless I create `/the/path/meson.build`:

```meson
subdir('subdir')
```

Can we use the following logic:

1. When `/the/path/meson.build` exists, use it.
2. When `user_edit_directory/meson.build` exists, use it.

It will be useful a git repo is not a meson projects, however, one or more subdir of it is a meson project, and other subidr store other things.

**Comment by JCWasmx86** [comment]
This is supported. Your editor just has to send the initialize request with a different path in the WorkspaceFolders (https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#initialize) . Not sure how you can do that, as I didn't test it yet.

**Reply by Freed-Wu** [reply]
Can we have a fallback `user_edit_directory/meson.build` for `/the/path/meson.build`? Can it be out of box without any extra configuration?

**Reply by JCWasmx86** [reply]
If you submit a patch and it's not that hacky, I would be happy to merge it. For example, adding support for reading a `.mesonlsp_help` file that contains:
```
some/path/
other/path/
```
These paths could be used as the new workspace folders. This is one possible solution, but not necessarily the only one.

**Reply by Freed-Wu** [reply]
> or example, adding support for reading a .mesonlsp_help file that contains:
> 
> some/path/
> other/path/

It looks similar as adding a `meson.build`:

```meson
subdir('some/path/')
subdir('other/path/')
```

**Reply by JCWasmx86** [reply]
Yes, but it was just a suggestion. I will review any patches regarding this topic
